hashicorp/terraform-provider-aws
# r/aws_ecs_task_definition: add track_latest attribute

### Description

This PR allows users to make the task definitions track always the latest one if `track_latest` is set to true.

The rationale for this Pull Request is that in some projects/environments/teams, Task Definitions can be updated outside Terraform. For example, a CI/CD process could bumps the Docker Tag version by creates a new TaskDefinition using the previous one as a base template, within the process, it might move the previous TaskDefinition to inactive. Because of this, when Terraform is executed again it cannot track the changes and forces a new resource creation.

With this Pull Request, we would allow users to track always the changes against the latest TaskDefinition, so, if all the fields are the same, Terraform won't promote any change.

In fact, while checking the code, I've seen that "family" is considered to be the ID of the resource, but when reading it, it was readed by "arn". The "family" doesn't contain a revision (version), while "arn" contains the revision.

### References
Here are some references to people updating the version outside terraform:
* https://github.com/hashicorp/terraform-provider-aws/issues/20121
* https://github.com/hashicorp/terraform-provider-aws/issues/632


### Output from Acceptance Testing


> Regarding the acceptance tests, I'd need some guidance from someone, please.
I'd like to write the tests, but I'm not sure how sould I write them.

The test scenario would be:

1. Create a TaskDefinition with Terraform
2. Create a new TaskDefinition based with the previous one, and then move the previous one to inactive
3. Plan Terraform changes again and should not show any change.

If someone thinks this pull is interesting, and wants to guide me with the tests, I'd highly appreciate!

```
$ make testacc TESTS=TestAccXXX PKG=ec2

...
```
